secant/Features/Scan/UIKitBridge/QRCodeScanView.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  QRCodeScanView.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 16.05.2022.
+//
+
+import Foundation
+import UIKit
+import SwiftUI
+
+struct QRCodeScanView: UIViewRepresentable {
+    var rectOfInterest = CGRect(x: 0.0, y: 0.0, width: 1.0, height: 1.0)
+    let onQRScanningDidFail: () -> Void
+    let onQRScanningSucceededWithCode: (String?) -> Void
+    let onQRScanningDidStop: () -> Void
+
+    public func makeUIView(context: UIViewRepresentableContext<QRCodeScanView>) -> ScanUIView {
+        let view = ScanUIView()
+        view.continueVideoSessionAfterScanning = true
+        view.rectOfInterest = rectOfInterest
+        view.onQRScanningDidFail = onQRScanningDidFail
+        view.onQRScanningSucceededWithCode = onQRScanningSucceededWithCode
+        view.onQRScanningDidStop = onQRScanningDidStop
+        return view
+    }
+    
+    public func updateUIView(_ uiView: ScanUIView, context: UIViewRepresentableContext<QRCodeScanView>) { }
```

Comment:
When a value of the UIViewRepresentable is changed, it has been my experience that the object underneath won't be recreated. Although most of the UIViewRepresentable code samples you can see will have this method empty, it's an odd implementation. You should always upgrade the underlying UIView accordingly form the context to avoid the view not being responsive or taking invalid values.

Reply:
What would you update? Like you said in many scenarios the method is empty but when it's not there are view properties updated. We don't hold any here. Once the view is created, we no longer update any values in it. The context.environment is not providing us any meaningful information.

The typical usage is to hold binding text for example and update its value (in case of search for example).

Reply:
oh, I missed the fact that all the members were constants. Yet I'm not sure whether a refresh on the struct at a SwiftUI DSL will recreate the UIView underneath. 
Let's leave it as it is and I'll further research this. I've burned with milk when it comes to UIViewRepresentables.